src/SFA.DAS.AssessorService.Web.Staff/Controllers/Roatp/AddRoatpOrganisationController.cs

```diff
@@ -0,0 +1,163 @@
+            }
+
+            model.ProviderTypes = await _apiClient.GetProviderTypes();
+
+            _sessionService.SetAddOrganisationDetails(model);
+
+            return View("~/Views/Roatp/AddOrganisation.cshtml", model);
+        }
+
+        [Route("enter-details")]
+        public async Task<IActionResult> AddOrganisationDetails(AddOrganisationViewModel model)
+        {
+            var validationMessages = _validator.ValidateProviderType(model.ProviderTypeId);
+            if (validationMessages.Any())
+            {
+                model.ValidationErrors = new List<string>();
+                model.ValidationErrors.AddRange(validationMessages);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                return View("~/Views/Roatp/AddOrganisation.cshtml", model);
+            }
+
+            model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+
+            var sessionModel = _sessionService.GetAddOrganisationDetails(model.OrganisationId);
+            if (sessionModel.ProviderTypeId != model.ProviderTypeId)
+            {
+                model.OrganisationTypeId = 0;
+            }
+
+            _sessionService.SetAddOrganisationDetails(model);
+
+            return View("~/Views/Roatp/AddOrganisationDetails.cshtml", model);
+        }
+
+        [Route("confirm-details")]
+        public async Task<IActionResult> AddOrganisationPreview(AddOrganisationViewModel model)
+        {
+            model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+            model.ProviderTypes = await _apiClient.GetProviderTypes();
+
+            var validationMessages = _validator.ValidateOrganisationDetails(model);
+            if (validationMessages.Any())
+            {
+                model.ValidationErrors = new List<string>();
+                model.ValidationErrors.AddRange(validationMessages);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                return View("~/Views/Roatp/AddOrganisationDetails.cshtml", model);
+            }
+
+            _sessionService.SetAddOrganisationDetails(model);
+
+            return View("~/Views/Roatp/AddOrganisationPreview.cshtml", model);
+        }
+
+        [Route("successfully-added")]
+        public async Task<IActionResult> CreateOrganisation(AddOrganisationViewModel model)
+        {
+            var request = CreateAddOrganisationRequestFromModel(model);
+
+            var success = await _apiClient.CreateOrganisation(request);
+
+            if (!success)
+            {
+                model.OrganisationTypes = await _apiClient.GetOrganisationTypes(model.ProviderTypeId);
+                model.ProviderTypes = await _apiClient.GetProviderTypes();
+                model.ValidationErrors = new List<string>
+                {
+                    $"An error occurred when adding the organisation '{model.LegalName}'.<br/> Please try again later."
```

Comment:
We could do with these being in a central resx maybe? That more for tech debt, so don't resolve here.

Reply:
Will be picked up in APR-233